From 9ef8ed28322a1282d6d5cb3cc9dcfe9e8fa9c23c Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Sun, 21 Jul 2019 16:34:15 +0200
Subject: [PATCH 345/386] net: mvpp2: fix TX done tasklet data

The TX done tasklet expects struct net_device
instead of struct mvpp2_port as an argument. Fix that.

Change-Id: I77ffc78a6446b1657ca171362d800b505552fe24
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12828
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 48c2d14c765b..fc069c07e91e 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -3614,7 +3614,7 @@ static void mvpp2_tx_done_init_on_open(struct mvpp2_port *port, bool open)
 			port_pcpu->guard_timer_scheduled = true;
 			tasklet_init(&port_pcpu->tx_done_tasklet,
 				     mvpp2_tx_done_proc_cb,
-				     (unsigned long)port);
+				     (unsigned long)port->dev);
 		}
 	}
 	return;
-- 
2.17.1

